components/places/src/ffi.rs

```diff
@@ -113,6 +113,39 @@ impl PlacesConnection {
             )
         })
     }
+
+    fn get_top_frecent_site_infos(
+        &self,
+        num_items: i32,
+        threshold_option: FrecencyThresholdOption,
+    ) -> Result<Vec<TopFrecentSiteInfo>> {
+        self.with_conn(|conn| {
+            crate::storage::history::get_top_frecent_site_infos(
+                conn,
+                num_items,
+                threshold_option.value(),
+            )
+        })
+    }
+}
+
+pub struct TopFrecentSiteInfo {
+    pub url: Url,
+    pub title: Option<String>,
+}
+
+pub enum FrecencyThresholdOption {
+    None,
+    SkipOneTimePages,
+}
+
+impl FrecencyThresholdOption {
+    fn value(&self) -> i64 {
+        match self {
+            FrecencyThresholdOption::None => 0,
+            FrecencyThresholdOption::SkipOneTimePages => 101,
```

Comment:
I'm a little iffy about those magic numbers... But they were magic numbers in the kotlin file before...

Reply:
I was hoping to find a useful comment [in desktop](https://searchfox.org/mozilla-central/source/browser/components/newtab/lib/TopSitesFeed.jsm#87), but that didn't pan out as I had hoped. 😬

Reply:
Lol I'll define the value as a constant at the top of the file, add some comments and a link to desktop at the very minimum, maybe that's enough for now 🤷

Reply:
I *think* it relates to the constants [here](https://github.com/mozilla/application-services/blob/main/components/places/src/frecency.rs#L52,L59) and is basically saying "does this visit have something other than the smallest possible frecency?". Or something like that 😅